src/kvtree_io.c

```diff
@@ -36,6 +36,11 @@
 /* gettimeofday */
 #include <sys/time.h>
 
+/* pthread_mutex */
+#include <pthread.h>
+
+pthread_rwlock_t file_lock = PTHREAD_RWLOCK_INITIALIZER;
+
```

Comment:
Let's drop the pthread stuff for now.

Reply:
The mutex is there to make the file locking functions thread safe.  Why remove it?

Reply:
The flip question to that is why do we just cherry pick one function from kvtree and make it thread safe, when the others are not?  Also, we don't have any code that requires this to be thread safe.

It's cleaner to just state that none of kvtree is thread safe, drop libpthread as a dependency, and put all responsibility on the user of kvtree to lock/unlock if they need it.

Reply:
Ok, I'll remove the mutex then, and adjust the test case to `fork()` instead of use pthreads.  That should exercise the `flock()/fnctl()` part of the kvtree file-locking functions.